distributed/bokeh/tests/test_application.py

```diff
@@ -29,3 +29,15 @@ def test_BokehWebInterface(loop):
                 sleep(0.01)
     with pytest.raises(Exception):
         response = requests.get('http://127.0.0.1:8787/status/')
+
+
+def test_bokeh_shutsdown_without_cluster___del__(loop):
+    c = LocalCluster(2, loop=loop, scheduler_port=0,
+                     services={('http', 0): HTTPScheduler},
+                     diagnostic_port=None)
+    proc = c.diagnostics.process
+    # don't run the del, as it isn't ever run in python < 3.5 due to cycles
+    c.__del__ = lambda self: None
+    del c
+    sleep(0.1)
+    assert proc.poll() is not None
```

Comment:
In general trusting `sleep`s on travis.ci is tough.  Instead I recommend the following pattern:

```python
start = time()
while proc.poll() is None:
    sleep(0.01)
    assert time() < start + 5
```